src/request/swap-iframe/SwapIFrameApi.js

```diff
@@ -0,0 +1,283 @@
+            psbt.addOutputs(outputs);
+
+            // Sign
+            const keyPairs = privateKeys.btc.map(privateKey => BitcoinJS.ECPair.fromPrivateKey(
+                // @ts-ignore Argument of type 'import("...").Buffer' is not assignable to parameter of type 'Buffer'.
+                BitcoinJS.Buffer.from(privateKey, 'hex'),
+            ));
+            for (const keyPair of keyPairs) {
+                psbt.signAllInputs(keyPair);
+            }
+
+            // Verify that all inputs are signed
+            if (!psbt.validateSignaturesOfAllInputs()) {
+                throw new Error('Invalid or missing signature(s) for BTC transaction.');
+            }
+
+            // Finalize
+            psbt.finalizeAllInputs();
+
+            // Extract tx
+            const tx = psbt.extractTransaction();
+
+            /** @type {KeyguardRequest.SignedBitcoinTransaction} */
+            result.btc = {
+                transactionHash: tx.getId(),
+                raw: tx.toHex(),
+            };
+        }
+
+        if (request.redeem.type === 'NIM') {
+            await loadNimiq();
+
+            const privateKey = new Nimiq.PrivateKey(Nimiq.BufferUtils.fromHex(privateKeys.nim));
+            const publicKey = Nimiq.PublicKey.derive(privateKey);
+
+            const transaction = this.parseTransaction(request.redeem);
+
+            const signature = Nimiq.Signature.create(privateKey, publicKey, transaction.serializeContent());
+
+            /** @type {KeyguardRequest.SignatureResult} */
+            result.nim = {
+                publicKey: publicKey.serialize(),
+                signature: signature.serialize(),
+            };
+        }
+
+        if (request.redeem.type === 'BTC') {
+            const inputs = this.parseInputs(request.redeem.inputs.map(input => ({
+                ...input,
+                // Dummy keyPath so the parsing doesn't fail.
+                // The keyPath is not used in the iframe, as the key at the path was already derived
+                // (and stored) in the top-level sign-swap request.
+                keyPath: 'm/84\'/0',
+            })));
+
+            /** @type {KeyguardRequest.BitcoinTransactionOutput} */
+            const output = (this.parseOutput(request.redeem.changeOutput, false, 'redeem.changeOutput'));
+
+            // Construct transaction
+            const psbt = new BitcoinJS.Psbt({ network: BitcoinUtils.Network });
+
+            // Add inputs
+            // @ts-ignore Argument of type 'Uint8Array' is not assignable to parameter of type 'Buffer'.
+            psbt.addInputs(inputs);
+            // Add outputs
+            psbt.addOutput(output);
+
+            // Sign
+            const keyPair = BitcoinJS.ECPair.fromPrivateKey(
+                // @ts-ignore Argument of type 'import("...").Buffer' is not assignable to parameter of type 'Buffer'.
+                BitcoinJS.Buffer.from(privateKeys.btc[0], 'hex'),
+            );
+            psbt.signInput(0, keyPair);
```

Comment:
Is there only ever going to be 1 input?

Reply:
Yes, BTC HTLCs are only funded with one UTXO.

Doesn't _have_ to, it's just a BTC address, can have multiple UTXOs as well. But the way Fastspot works, there's only one.